tests/integrations/aws_lambda/test_aws.py

```diff
@@ -275,14 +276,5 @@ def test_handler(event, context):
         b'{"foo": "bar"}',
     )
     expected_text = "WARNING : Function is expected to get timed out. Configured timeout duration = 4 seconds"
-    if not events:
-        # In case of Python 2.7 runtime environment
-        log_result = (base64.b64decode(response["LogResult"])).decode("utf-8")
-        assert expected_text in log_result
-    else:
-        # In case of Python 3.6, 3.7 & 3.8 runtime environments
-        (event,) = events
-        assert event["level"] == "error"
-        (exception,) = event["exception"]["values"]
-        assert exception["type"] == "Exception"
-        assert exception["value"] == expected_text
+    log_result = (base64.b64decode(response["LogResult"])).decode("utf-8")
```

Comment:
I guess I am still confused as to what we're asserting here. It seems we are not expecting any timeout event at all?

Reply:
@untitaker 
Here, we are asserting with the exception message that we are sending when the ServerlessTimeoutWarning custom exception is raise in case of timeout error. If the exception is raised it'll be there in the log result. 
Also, timeout event is not coming even after the solution that you suggested. I've applied it is this manner:-

`def run(thread_hub, configured_timeout):
    with thread_hub:
        try:
            raise ServerlessTimeoutWarning(
                    "WARNING : Function is expected to get timed out. Configured timeout duration = {} seconds.".format(
                        configured_timeout
                    )
                )
        except Exception:
            client = thread_hub.client
            exc_info = sys.exc_info()
            event, hint = event_from_exception(
                exc_info,
                client_options=client.options,
                mechanism={"type": "threads", "handled": False},
            )
            thread_hub.capture_event(event, hint=hint)
            reraise(*exc_info)`

And, while starting a new thread:

`thread_hub = Hub(Hub.current)

tr = threading.Thread(target=run, args=[thread_hub, configured_timeout])
tr.start()`

This does generate timeout warning and the mechanism I provided in above code appears on Sentry dashboard, but it still doesn't show the stacktrace which was required and neither does the events data come in case f the automation test cases.

Reply:
`thread_hub = Hub.current` ought to be correct.

I can't really tell what would be wrong but as-is the testcase is wrong. Y'all have to debug this. If it makes things easier you can also try `hub.capture_message` without any exception.

It's fine if there's no stacktrace. The only useful stacktrace comes from the sigalrm approach we discussed in Slack afaik.

Reply:
So you mean that event should come irrespective of whatever be the test case? Maybe, it is related to the HttpTransport we are using and we might need to modify it to send event in case of timeout error.

Reply:
I mean the test case right here is supposed to assert that there is an event, right? but right now it's not asserting that.

I don't think the transport is the issue tbh

Reply:
Okay, I'll debug through this issue and see why event data is not coming and post the updates.

Reply:
@untitaker I've debugged this issue. As I suspected earlier as well, the events are not coming because of the _send_event() method defined inside LAMBDA_PRELUDE of test_aws.py. The events =basically are not coming because of the delay of 1 second that is kept there.
Also, the reason why events are not coming after 1 second delay is because of the same reason timeout warning needs to be generated a litter before the actual timeout.

Is it very necessary to keep that delay ? can we remove that delay ?